auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -404,10 +410,12 @@ public TypeSpec createTypeAdapter(Context context, ClassName className, ClassNam
         .superclass(superClass)
         .addFields(adapters.values())
         .addFields(defaultValueFields.values())
+        .addField(FieldSpec.builder(Gson.class, "gson", PRIVATE).build())
```

Comment:
As opposed to storing a reference to the Gson instance, it would be cleaner to just store the result of `fieldNamingStrategy()` here.  That's all it's used for in the `renameField()` method, and storing that would reduce the number of method call in the adapter, since we'd have a cached value.  

The FieldNamingStrategy is immutable, so it won't change at runtime unless the user fetches a new instance of Gson, thereby creating a new TypeAdapter.